samza-core/src/main/scala/org/apache/samza/metrics/JvmMetrics.scala

```diff
@@ -153,4 +163,16 @@ class JvmMetrics(val registry: MetricsRegistry) extends MetricsHelper with Runna
     gThreadsTimedWaiting.set(threadsTimedWaiting)
     gThreadsTerminated.set(threadsTerminated)
   }
+
+  private def updateOperatingSystemMetrics {
+    if (osMXBean.isInstanceOf[OperatingSystemMXBean]) {
+      val operatingSystemMXBean = osMXBean.asInstanceOf[OperatingSystemMXBean]
+      gProcessCpuUsage.set(operatingSystemMXBean.getProcessCpuLoad * PCT)
+      gSystemCpuUsage.set(operatingSystemMXBean.getSystemCpuLoad * PCT)
+
+      if (osMXBean.isInstanceOf[UnixOperatingSystemMXBean]) {
```

Comment:
Can we consolidate these `isInstanceOf` checks to a single flag `isUnix` and set it during construction?